permissions/src/iosMain/kotlin/location/LocationPermissionManager.kt

```diff
@@ -0,0 +1,102 @@
+import com.splendo.kaluga.permissions.Permission
+import com.splendo.kaluga.permissions.PermissionManager
+import com.splendo.kaluga.permissions.PermissionState
+import platform.CoreLocation.CLAuthorizationStatus
+import platform.CoreLocation.CLLocationManager
+import platform.CoreLocation.CLLocationManagerDelegateProtocol
+import platform.Foundation.NSBundle
+import platform.darwin.NSObject
+
+actual class LocationPermissionManager(
+    private val bundle: NSBundle,
+    actual val location: Permission.Location,
+    stateRepo: LocationPermissionStateRepo
+) : PermissionManager<Permission.Location>(stateRepo) {
+
+    private val locationManager = CLLocationManager()
+    private val authorizationStatus = {
+        CLLocationManager.authorizationStatus().toAuthorizationStatus(location.background)
+    }
+
+    private val delegate = object : NSObject(), CLLocationManagerDelegateProtocol {
+
+        override fun locationManager(manager: CLLocationManager, didChangeAuthorizationStatus: CLAuthorizationStatus /* = kotlin.Int */) {
+            val locationPermissionManager = this@LocationPermissionManager
+            IOSPermissionsHelper.handleAuthorizationStatus(didChangeAuthorizationStatus.toAuthorizationStatus(locationPermissionManager.location.background), locationPermissionManager)
+        }
+
+    }
+
+    override suspend fun requestPermission() {
+        val locationDeclarations = mutableListOf("NSLocationWhenInUseUsageDescription")
+        if (location.background) {
+            locationDeclarations.addAll(listOf("NSLocationAlwaysAndWhenInUseUsageDescription", "NSLocationAlwaysUsageDescription"))
+        }
+        if (IOSPermissionsHelper.checkDeclarationInPList(bundle, *locationDeclarations.toTypedArray()).isEmpty()) {
+            if (location.background)
+                locationManager.requestAlwaysAuthorization()
+            else
+                locationManager.requestWhenInUseAuthorization()
+        } else {
+            revokePermission(true)
+        }
+    }
+
+    override suspend fun initializeState(): PermissionState<Permission.Location> {
+        return IOSPermissionsHelper.getPermissionState(authorizationStatus(), this)
+    }
+
+    override suspend fun startMonitoring(interval: Long) {
+        locationManager.delegate = delegate
+    }
+
+    override suspend fun stopMonitoring() {
+        locationManager.delegate = null
+    }
+
+}
+
+actual class LocationPermissionManagerBuilder(private val bundle: NSBundle = NSBundle.mainBundle) : BaseLocationPermissionManagerBuilder {
+
+    override fun create(location: Permission.Location, repo: LocationPermissionStateRepo): LocationPermissionManager {
+        return LocationPermissionManager(bundle, location, repo)
+    }
+
+}
+
+private fun CLAuthorizationStatus.toAuthorizationStatus(background: Boolean): IOSPermissionsHelper.AuthorizationStatus {
+    return when(this) {
+        0 -> IOSPermissionsHelper.AuthorizationStatus.NotDetermined
```

Comment:
I believe there is constants in CoreLocation library for that

Reply:
You'd expect that, but unlike the other Libraries, there is no such thing as CLAuthorizationStatusAuthorized or something. Not sure why but its just not there

Reply:
Then you can use our own, it's called `CLAuthorizationStatusKotlin` inside LocationFlowable

Reply:
Done. Moved the existing enum out of location and into permissions, so its reusable when I start the location refactor